asset-transfer-secured-agreement/application-javascript/app.js

```diff
@@ -0,0 +1,563 @@
+				// Since the Org2 is not the owner, this will fail
+				console.log(`${GREEN}--> Submit Transaction: ChangePublicDescription ${assetKey}, as Org2 - endorse by Org2${RESET}`);
+				transaction = contractOrg2.createTransaction('ChangePublicDescription');
+				transaction.setEndorsingOrganizations(org2);
+				await transaction.submit(assetKey, `Asset ${assetKey} owned by ${org2} is NOT for sale`);
+				console.log(`${RESET}*** Failed: Org2 is not the owner and this should have failed${RESET}`);
+			} catch (updateError) {
+				console.log(`*** Success: ChangePublicDescription has failed endorsememnt by Org2 - ${updateError}`);
+			}
+
+			try {
+				// This is an update to the public state and requires only the owner to endorse.
+				// Since the Org2 is not the owner, this will fail
+				console.log(`${GREEN}--> Submit Transaction: ChangePublicDescription ${assetKey}, as Org2 - endorse by Org1${RESET}`);
+				transaction = contractOrg2.createTransaction('ChangePublicDescription');
+				transaction.setEndorsingOrganizations(org1);
+				await transaction.submit(assetKey, `Asset ${assetKey} owned by ${org2} is NOT for sale`);
+				console.log(`${RESET}*** Failed: Org2 is not the owner and this should have failed${RESET}`);
+			} catch (updateError) {
+				console.log(`*** Success: ChangePublicDescription has failed endorsement by Org1 - ${updateError}`);
+			}
+
+			// read the public details by both orgs
+			await readAsset(assetKey, org1, contractOrg1, contractOrg2);
+
+			try {
+				// Agree to a sell by Org1
+				const asset_price = {
+					asset_id: assetKey,
+					price: 110,
+					trade_id: randomNumber.toString()
+				};
+				const asset_price_string = JSON.stringify(asset_price);
+				console.log(`${GREEN}--> Submit Transaction: AgreeToSell, ${assetKey} as Org1 - endorsed by Org1${RESET}`);
+				transaction = contractOrg1.createTransaction('AgreeToSell');
+				transaction.setEndorsingOrganizations(org1);
+				transaction.setTransient({
+					asset_price: Buffer.from(asset_price_string)
+				});
+				await transaction.submit(assetKey);
+				console.log(`*** Result: committed, Org1 has agreed to sell asset ${assetKey} for 110`);
+			} catch (sellError) {
+				console.log(`${RED}*** Failed: AgreeToSell - ${sellError}${RESET}`);
+			}
+
+			try {
+				// check the private information about the asset from Org2
+				// Org1 would have to send Org2 these details so the hash of the
+				// details may be checked by the chaincode.
+				const asset_properties = {
+					object_type: 'asset_properties',
+					asset_id: assetKey,
+					color: 'blue',
+					size: 35,
+					salt: Buffer.from(randomNumber.toString()).toString('hex')
+				};
+				const asset_properties_string = JSON.stringify(asset_properties);
+				console.log(`${GREEN}--> Submit Transaction: VerifyAssetProperties, ${assetKey} as Org2 - endorsed by Org2${RESET}`);
+				console.log(`${asset_properties_string}`);
+				transaction = contractOrg2.createTransaction('VerifyAssetProperties');
```

Comment:
VerifyAssetProperties is a query only.
It should not be submitted for commit.

Reply:
done